### PR TITLE
[Merged by Bors] - EnvironmentMapLight support for WebGL2

### PR DESCRIPTION
**Pull request by JMS55**
# Objective

- Fix the environment map shader not working under webgl due to textureNumLevels() not being supported
- Fixes https://github.com/bevyengine/bevy/issues/7722

## Solution

- Instead of using textureNumLevels(), put an extra field in the GpuLights uniform to store the mip count

### CODE REVIEW

File: crates/bevy_pbr/src/render/mesh.rs [hunk 1]
```diff
@@ -651,6 +652,9 @@ impl SpecializedMeshPipeline for MeshPipeline {
         let mut shader_defs = Vec::new();
         let mut vertex_attributes = Vec::new();
 
+        #[cfg(feature = "webgl")]
+        shader_defs.push("WEBGL".into());
+
```
Comment (mockersf) [inline crates/bevy_pbr/src/render/mesh.rs:657]: could you move that shaderdef to https://github.com/bevyengine/bevy/blob/64b0f19a6547071b54f7479deb55a5494f0fe204/crates/bevy_render/src/render_resource/pipeline_cache.rs#L187-L191?

**Comment by superdump** [comment]
I suppose as the impact is minimal, just always using either a shader def that sets a value or a uniform would serve all use cases.

File: crates/bevy_pbr/src/render/mesh.rs (hunk 1 above)
Comment (JMS55) [reply crates/bevy_pbr/src/render/mesh.rs:657]: Yes, that makes sense.

**Comment by JMS55** [comment]
@superdump shader def is a bit hard, because we need a non-constant value, which means we need to reserve ~4 whole bits of the pipeline key just to be able to store the value.

I can switch to _always_ using a uniform though, instead of making it a fallback like this. Do you prefer that, or how it is now?

**Comment by superdump** [comment]
I’d say go for always using a uniform. 👍

**Review by superdump — APPROVED** [review]
bors r+

**Pull request closed without merging**